onadata/libs/pagination.py

```diff
@@ -57,3 +62,73 @@ def paginate_queryset(self, queryset, request, view, count=None):
 
         self.request = request
         return list(self.page)
+
+
+def generate_pagination_headers(
+    request: Request, no_of_records: int, current_page_size: Union[int, str],
+    current_page: Union[int, str]
+) -> dict:
+    url = urlparse(request.build_absolute_uri())
+    base_url = f"{url.scheme}://{url.netloc}{url.path}?"
+    if url.query:
+        query_params = parse_qs(url.query)
+        query_string = None
+        for param_key, param_values in query_params.items():
+            if param_key not in ['page_size', 'page']:
+                param_value = ','.join(param_values)
+                query = f'{param_key}={param_value}&'
```

Comment:
Ii've tested this locally, after looping through all the query params passed with the request, the result url seems to appear as such
``` code
'http://localhost:8000/api/v1/messaging?target_type=xform&target_id=42&'
```
The trailing `&`, is that something that may be a proble in future?